### ISSUE TITLE
All Custom Dimensions for website 2 in scope 'visit' are already used.

### ISSUE THREAD

**Issue by tsteur** [matomo-org/plugin-GoogleAnalyticsImporter#120]
Import always fails with this message and doesn't seem to proceed @diosmosis :

```
Error: {"message":"All Custom Dimensions for website 2 in scope 'visit' are already used.","file":"\/var\/www\/html\/plugins\/CustomDimensions\/Dimension\/Index.php","line":35,"backtrace":" on \/var\/www\/html\/plugins\/CustomDimensions\/Dimension\/Index.php(35)\n#0 \/var\/www\/html\/plugins\/CustomDimensions\/API.php(116): Piwik\\Plugins\\CustomDimensions\\Dimension\\Index->getNextIndex('2', 'visit')\n#1 \/var\/www\/html\/plugins\/GoogleAnalyticsImporter\/Importer.php(291): Piwik\\Plugins\\CustomDimensions\\API->configureNewCustomDimension('2', 'XXX', 'visit', true, Array, true)\n#2
```

I suppose the problem is that they might have more dimensions on GA than configured in Matomo. Not sure what the best flow be as we probably can't easily notify the user about this.

**Comment by tsteur** [comment]
Note the error message is specific to Cloud but it would also happen in On-Premise I suppose when there is no more slot available. I reckon it be a different error message though

**Comment by diosmosis** [comment]
Working on a fix, will ask in the UI if the user wants to continue ignoring missing custom dimensions, while import itself will just ignore if a custom dimension slot is unavailable.

**Comment by diosmosis** [comment]
Fixed by #123
